tests/src/rules/no-extraneous-dependencies.js

```diff
@@ -378,7 +380,7 @@ ruleTester.run('no-extraneous-dependencies', rule, {
   ],
 });
 
-describe('TypeScript', () => {
+(semver.satisfies(typescriptResolverPkg.version, '>=2.1.0') ? describe : describe.skip)('TypeScript', () => {
```

Comment:
this means the files in here won't be tested on v1 (v1.0 or v1.1) of the TS resolver. can we preserve that?

Reply:
Yes but ... This only works with >=2.0.0 of the resolver because earlier versions didn't look in `@types` packages, so testing against them doesn't confirm any special type-only treatment because they return the same thing as the node resolver? Like we exclude typescript-eslint-parser because it predates type-only imports.

I can see adding a "Legacy TypeScript resolver" test and running that in case of an earlier version. I think I'd prefer not, and keep the tests focused on the desired behavior (like we exclude typescript-eslint-parser). Maybe there's a reason it's important though? I'm happy either way!

Reply:
Mainly semver - things that work should keep working, and the tests ensure that's the case.

Reply:
Fair enough. I've now added a test to confirm that the behavior of earlier versions is unchanged?